src/EA.Weee.DataAccess.Tests.DataAccess/StoredProcedure/GetAllAatfSentOnDataCsvTests.cs

```diff
@@ -0,0 +1,90 @@
+                @return.UpdateSubmitted(db.Model.AspNetUsers.First().Id, false);
+                var aatf = ObligatedWeeeIntegrationCommon.CreateAatf(db, @return.Organisation);
+
+                db.WeeeContext.Returns.Add(@return);
+
+                await db.WeeeContext.SaveChangesAsync();
+
+                var results = await db.StoredProcedures.GetAllAatfSentOnDataCsv(2019, string.Empty, string.Empty, null, null);
+
+                results.Tables[0].Rows.Count.Equals(0);
+                results.Tables[1].Rows.Count.Equals(0);
+            }
+        }
+
+        [Fact]
+        public async Task Execute_GivenWeeeSentOnData_ReturnsWeeeSentOnAatfDataShouldBeCorrect()
+        {
+            using (var db = new DatabaseWrapper())
+            {
+                var @return = CreateSubmittedReturn(db);
+                var aatf = ObligatedWeeeIntegrationCommon.CreateAatf(db, @return.Organisation);
+                var weeeSentOn = new EA.Weee.Domain.AatfReturn.WeeeSentOn(ObligatedWeeeIntegrationCommon.CreateAatfAddress(db), ObligatedWeeeIntegrationCommon.CreateAatfAddress(db), aatf, @return);
+                var weeeSentOnAmounts = new List<WeeeSentOnAmount>()
+                {
+                    new Domain.AatfReturn.WeeeSentOnAmount(weeeSentOn, 1, 1, 2),
+                    new Domain.AatfReturn.WeeeSentOnAmount(weeeSentOn, 2, 3, 4)
+                };
+
+                db.WeeeContext.Returns.Add(@return);
+                db.WeeeContext.ReturnAatfs.Add(new ReturnAatf(aatf, @return));                
+                db.WeeeContext.WeeeSentOn.Add(weeeSentOn);
+                db.WeeeContext.WeeeSentOnAmount.AddRange(weeeSentOnAmounts);
+
+                await db.WeeeContext.SaveChangesAsync();
+
+                db.WeeeContext.ReturnReportOns.Add(new ReturnReportOn(@return.Id, 2));
+
+                await db.WeeeContext.SaveChangesAsync();
+
+                var results = await db.StoredProcedures.GetAllAatfSentOnDataCsv(2019, string.Empty, string.Empty, null, null);
+
+                Assert.NotNull(results);
+                var data = from x in results.Tables[0].AsEnumerable()
+                           where x.Field<string>("Name of AATF") == aatf.Name
+                           select x;
+                data.AsQueryable().Count().Should().Be(28);
+
+                var dataB2B = from x in results.Tables[0].AsEnumerable()
+                              where x.Field<string>("Name of AATF") == aatf.Name && x.Field<string>("Obligation") == "B2B"
+                              select x;
+                dataB2B.AsQueryable().Count().Should().Be(14);
+
+                var dataB2C = from x in results.Tables[0].AsEnumerable()
+                              where x.Field<string>("Name of AATF") == aatf.Name && x.Field<string>("Obligation") == "B2C"
+                              select x;
+                dataB2C.AsQueryable().Count().Should().Be(14);
```

Comment:
Can some tests be added to test the stored procedure parameters

Reply:
Added